openwisp_controller/config/static/config/js/unsaved_changes.js

```diff
@@ -12,7 +12,9 @@
                     name == 'csrfmiddlewaretoken' ||
                     // ignore hidden inline helper fields
                     name.indexOf('__prefix__') >= 0 ||
-                    name.indexOf('root') === 0) {
+                    name.indexOf('root') === 0 ||
+                    // ignore device location field
+                    name.substr(0, 14) == 'devicelocation') {
```

Comment:
this partially solves the issue, but what if the user changes the device location and forgets to save?
Can you find a way to make that case work as well?